rotkehlchen/chain/ethereum/modules/eth2/decoder.py

```diff
@@ -94,11 +105,54 @@ def _decode_eth2_deposit_event(self, context: DecoderContext) -> DecodingOutput:
         )
         return DEFAULT_DECODING_OUTPUT
 
+    def _decode_eth2_consolidation_request(self, context: DecoderContext) -> DecodingOutput:
+        """Decode a request to consolidate validators.
+        Handles two types of requests:
+        - Self consolidation - changes the withdrawal credentials to 0x02 (accumulating validator)
+          See https://epf.wiki/?ref=blog.obol.org#/wiki/pectra-faq?id=q-i-have-a-validator-with-0x01-credentials-how-do-i-move-to-0x02
+        - Normal consolidation - consolidates the source validator into the target validator
+        """
+        if len(context.tx_log.data) != 116:  # This an anonymous tx log so can't check topic[0]
+            return DEFAULT_DECODING_OUTPUT
+
+        for event in context.decoded_events:
+            if (
+                event.address == CONSOLIDATION_REQUEST_CONTRACT and
+                event.event_type == HistoryEventType.SPEND and
+                event.event_subtype == HistoryEventSubType.NONE and
+                event.asset == A_ETH
+            ):
+                event.event_subtype = HistoryEventSubType.FEE
+                event.counterparty = CPT_ETH2
+                event.notes = f'Spend {event.amount} ETH as validator consolidation fee'
+
+        caller_address = bytes_to_address(b'\x00' * 12 + context.tx_log.data[:20])
+        source_public_key = Eth2PubKey(encode_hex(context.tx_log.data[20:68]))
+        target_public_key = Eth2PubKey(encode_hex(context.tx_log.data[68:116]))
+        target_idx_or_key = self._get_index_or_public_key(public_key=target_public_key)
+        return DecodingOutput(event=self.base.make_event_next_index(
+            tx_hash=context.transaction.tx_hash,
+            timestamp=context.transaction.timestamp,
+            event_type=HistoryEventType.INFORMATIONAL,
+            event_subtype=HistoryEventSubType.NONE,
+            asset=A_ETH,
+            amount=ZERO,
+            location_label=caller_address,
+            notes=(
+                f'Request to convert validator {target_idx_or_key} into an accumulating validator'
+                if source_public_key == target_public_key else
+                f'Request to consolidate validator {self._get_index_or_public_key(source_public_key)} into {target_idx_or_key}'  # noqa: E501
```

Comment:
question: can you make `_get_index_or_public_key` able to take a list of public keys. If they are in the database you do a single request instead of two and if you need to query beaconchain then `get_validator_data` allows to query using a list instead so you will just need to process a bit the output. In the best scenario you are saving an extra query to the db and in the worst case you save 1 request to beaconchain (that costs money under certain conditions)